libs/execution/src/lib/hooks/hook.ts

```diff
@@ -0,0 +1,46 @@
+// SPDX-FileCopyrightText: 2025 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { type Result } from '../blocks';
+import { type ExecutionContext } from '../execution-context';
+import { type IOTypeImplementation } from '../types';
+
+/** When to execute the hook.*/
+export type HookPosition = 'preBlock' | 'postBlock';
+
+export interface HookOptions {
+  /** Whether the pipeline should await the hooks completion. `false` if omitted.*/
+  blocking?: boolean;
+  /** Optionally specify one or more blocks to limit this hook to. If omitted, the hook will be executed on all blocks*/
+  blocktypes?: string | string[];
```

Comment:
`string[]` instead of `string | string[]` would be fine here I think. The attribute's name is plural, which is weird if it's a `string` I feel like.